### ISSUE TITLE
etcd Secrets blocks the Shoot namespace deletion

### ISSUE THREAD

**Issue by ialidzhikov** [gardener/gardener#2064]
**What happened**:
To be able to delete a Shoot.

**What you expected to happen**:

**How to reproduce it (as minimally and precisely as possible)**:
1. Create a Shoot `bar`.

2. Delete the Shoot `bar`.

3. Ensure that the Shoot namespace will hang in `Terminating`:
```
NAME                    STATUS        AGE
shoot--foo--bar         Terminating   7h16m
```

There are 3 Secrets with finalizers `druid.gardener.cloud/etcd-druid`
```
$ k -n shoot--foo--bar get secret
NAME               TYPE                DATA   AGE
ca-etcd            Opaque              2      7h18m
etcd-client-tls    kubernetes.io/tls   3      7h18m
etcd-server-cert   kubernetes.io/tls   3      7h18m
```

No etcd CR.
```
$ k -n shoot--foo--bar get etcd
No resources found in shoot--foo--bar namespace.
```

**Anything else we need to know?**:

**Environment**:

- Gardener version: master
- Kubernetes version (use `kubectl version`):
- Cloud provider or hardware configuration:
- Others:

**Comment by georgekuruvillak** [comment]
Fixed in this [PR](https://github.com/gardener/etcd-druid/pull/27).  Ill release a hotfix for druid.

**Comment by ialidzhikov** [comment]
Thank you @georgekuruvillak !

**Comment by rfranzke** [comment]
Fixed with druid v0.1.5, merged with #2065
